### PR TITLE
Remove hooks lifting fns, add schemas

### PR DESCRIPTION
**Pull request by frenchy64**
Close https://github.com/threatgrid/iroh/issues/4517

These lifting functions are too difficult to use correctly and have outlived their usefulness.

This PR in particular makes `apply-hooks` take a map, rather than varargs.

<a name="qa">[§](#qa)</a> QA
============================

No QA is needed.

<a name="release-notes">[§](#release-notes)</a> Release Notes
=============================================================

```
intern: Remove hooks lifting fns, add schemas
```

<a name="squashed-commits">[§](#squashed-commits)</a> Squashed Commits
======================================================================

### CODE REVIEW

File: src/ctia/schemas/services.clj [hunk 1]
```diff
@@ -0,0 +1,13 @@
+(ns ctia.schemas.services
+  "Schemas for functions in our Trapperkeeper service
+  graph that are provided by external libraries"
+  (:require [schema.core :as s]))
+
+(s/defschema ConfigServiceFns
+  "The built-in Trapperkeeper configuration service.
+
+  https://github.com/puppetlabs/trapperkeeper/blob/master/documentation/Built-in-Configuration-Service.md#trapperkeepers-built-in-configuration-service"
```
Comment (vemv) [inline src/ctia/schemas/services.clj:9]: You can hit `y` in github to get an immutable link

File: project.clj [hunk 2]
```diff
@@ -197,9 +199,6 @@
              :prepush {:plugins [[yogsototh/lein-kibit "0.1.6-SNAPSHOT"]
                                  [lein-bikeshed "0.3.0"]]}}
 
-  :ci {:pedantic? :abort
```
Comment (vemv) [inline project.clj:200]: Ouch, thanks!

**Review by vemv — APPROVED** [review]
LGTM!

**Comment by frenchy64** [comment]
Thanks!

**Pull request merged**